### ISSUE TITLE
FR: Set Embedded Image as Custom Cover

### ISSUE THREAD

**Issue by CaptainLexington** [koreader/koreader#11501]
Right now one can take screenshots of the interior of a book and set them as a custom cover, but if you're just trying to capture a single image surrounded by flowing text, the best case scenario is that the image will be surrounded by a white border (especially because it is not possible to take a screenshot when viewing an image, as the screenshot gesture zooms instead).

I'd like to see a "Set As Cover" button when viewing an image close-up.

The primary use-case in my mind here is an omnibus ebook containing many different books in a series or by the same author. Usually the cover of the individual book is embedded at the beginning of that section of the text, and it would be nice to begin reading it by setting it as the cover in KOReader, without having to go back to calibre, update the cover, and reimport it. I want the custom cover to reflect my reading experience, and the "offiicial" cover from within calibre to reflect the ebook as a whole.

This feature was originally requested in #8889, and it was closed because Custom Covers had not yet been contemplated. One of those suggestions has been implemented since, in #11193

**Comment by Frenzie** [comment]
You're free to comment on closed issues btw. 

I will say that #11193 is much simpler, because there's already an image file that you can point to. With an image in a book the question is what to do about it. I suppose you'd have to extract it.

**Comment by CaptainLexington** [comment]
Either that, or perhaps automatically take a screenshot cropped or bounded to the size of the image?

**Comment by hius07** [comment]
> it is not possible to take a screenshot when viewing an image, as the screenshot gesture zooms instead

Simultaneous taps in two diagonal corners.

**Comment by CaptainLexington** [comment]
> Simultaneous taps in two diagonal corners.
That would be enough if screenshots in image viewer automatically cropped marginal whitespace (which might already be desirable behavior apart from this use case?)

**Comment by CaptainLexington** [comment]
If the image and the device screen have different aspect ratios, the screenshot will have white space:

![ImageViewer_2024-02-28_105432](https://github.com/koreader/koreader/assets/1500233/c962c26d-750c-458d-b81d-de62c63fb160)

Here's the Door of Durin, padded top and bottom to fit my phone's screen. Setting this as the cover includes the padding:

![FileManager_2024-02-28_105554](https://github.com/koreader/koreader/assets/1500233/80c069c3-bf8c-4c80-9f1c-07c229f3bfe9)

**Comment by Frenzie** [comment]
Right, of course. Indeed, I don't think you'll get around doing something custom for that.

**Comment by hius07** [comment]
I think, having https://github.com/koreader/koreader/issues/11460#issuecomment-1948360123, we would add a direct feature "Set image as custom book cover".
Any development in this direction is blocked with https://github.com/koreader/koreader/pull/11044 to avoid conflicts.
